### PR TITLE
[AC-1174] Bulk collection management

### PR DESCRIPTION
**Pull request by shane-melton**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Introduce a new dialog to bulk assign access to multiple collections at once.

Requires server changes in https://github.com/bitwarden/server/pull/3229.

## Code changes

- `web/src/app/vault/components/vault-items/*`
  - Add a new `bulkEditAccess` event type
  - Add the bulk `Access` menu option when viewing the organization vault that fires the new event type

- `web/src/app/vault/org-vault/vault.component.ts`
  - Add logic to watch for the new `bulkEditAccess` event type and opens the dialog

- `app/vault/org-vault/bulk-collections-dialog/bulk-collections-dialog.component.*`
  - Introduce the new standalone dialog component and template

- `web/src/app/vault/core/collection-admin.service.ts`
  - Introduce the new `bulkAssignAccess` method and a corresponding request model

- `web/src/app/vault/components/collection-dialog/collection-dialog.component.ts`
  - Move shared logic mapping logic out of component and into `access-selecto.model.ts` to be more easily shared with the new dialog component

## Screenshots

image

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: apps/web/src/app/vault/org-vault/bulk-collections-dialog/bulk-collections-dialog.component.html
Comment (differsthecat) [inline apps/web/src/app/vault/org-vault/bulk-collections-dialog/bulk-collections-dialog.component.html:1]: ❓ In [AC-1347](https://bitwarden.atlassian.net/browse/AC-1347), we are allowing editing of collections from the individual vault ([PR here](https://github.com/bitwarden/clients/pull/6081)). Will this functionality also be available in the individual vault as well? If so, the location of this dialog should probably be placed up a level outside of the `org-vault` folder.

File: apps/web/src/app/vault/components/vault-items/vault-item-event.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ export type VaultItemEvent =
   | { type: "viewAttachments"; item: CipherView }
   | { type: "viewCollections"; item: CipherView }
   | { type: "viewAccess"; item: CollectionView }
+  | { type: "bulkEditAccess"; items: CollectionView[] }
```
Comment (differsthecat) [inline apps/web/src/app/vault/components/vault-items/vault-item-event.ts:10]: ⛏️  Any opposition to making this event `bulkEditCollectionAccess` to be a little more specific? I have done this with a few other events here as collection events will also be listened to in the individual vault.

File: apps/web/src/app/vault/org-vault/bulk-collections-dialog/bulk-collections-dialog.component.html
Comment (shane-melton) [reply apps/web/src/app/vault/org-vault/bulk-collections-dialog/bulk-collections-dialog.component.html:1]: Great point, I'm just not sure if its supposed to be available in the individual vault, so I've reached out to product to find out more. If so, I'll make that change and move it out of the `org-vault` folder.

File: apps/web/src/app/vault/components/vault-items/vault-item-event.ts (hunk 1 above)
Comment (shane-melton) [reply apps/web/src/app/vault/components/vault-items/vault-item-event.ts:10]: No opposition here; I'm always for being more verbose/clear!

File: apps/web/src/app/vault/org-vault/bulk-collections-dialog/bulk-collections-dialog.component.html
Comment (shane-melton) [reply apps/web/src/app/vault/org-vault/bulk-collections-dialog/bulk-collections-dialog.component.html:1]: Its only meant for the org vault as an administrative function at this time; so I think it should likely stay in the `org-vault` for now unless there is another location you'd prefer.

File: apps/web/src/app/vault/components/vault-items/vault-item-event.ts (hunk 1 above)
Comment (shane-melton) [reply apps/web/src/app/vault/components/vault-items/vault-item-event.ts]: Done!

File: apps/web/src/app/vault/org-vault/bulk-collections-dialog/bulk-collections-dialog.component.html
Comment (differsthecat) [reply apps/web/src/app/vault/org-vault/bulk-collections-dialog/bulk-collections-dialog.component.html:1]: Nope, that works if it is an org vault only feature 🙂

**Review by eliykat — APPROVED** [review]
AC changes look good

**Pull request merged**